### PR TITLE
[FLINK-31793] remove dependency on flink-shaded guava for flink-connector-jdbc

### PR DESCRIPTION
**Pull request by WencongLiu**
**Remove the dependency on flink-shaded guava for flink-connector-jdbc.**

### CODE REVIEW

**Review by MartijnVisser — CHANGES_REQUESTED** [review]
@WencongLiu Thanks for this. Since it appears that Guava isn't already coming in via one of the dependencies, but instead we have to add it ourselves, we need to make sure that the added Guava dependency is shaded specifically for the Flink JDBC connector. 

Next to that, I think we should also use a newer version of Guava (Shaded used 30, you're including 19)

**Comment by MartijnVisser** [comment]
> Do you mean we need to shade some class of the Guava dependency in the pom? Currently there is a shade plugin but nothing is shaded.

Yes indeed. So that it's not `import com.google.common.collect.ImmutableMap;` but something like `import org.apache.flink.connector.jdbc.shaded.guava30.com.google.common.collect.ImmutableMap;`

**Comment by WencongLiu** [comment]
Done. Thanks for your review! @MartijnVisser

File: flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/utils/JdbcTypeUtil.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 import org.apache.flink.api.java.typeutils.ObjectArrayTypeInfo;
 import org.apache.flink.table.types.logical.LogicalTypeRoot;
 
-import org.apache.flink.shaded.guava30.com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap;
```
[the remark this answers is not in the document]
Comment (WencongLiu) [reply flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/utils/JdbcTypeUtil.java:29]: I think this import sentence will be replaced by `org.apache.flink.connector.jdbc.shaded.guava30` after being packaged by maven. Then other java projects which depend on jdbc-connector can import org.apache.flink.connector.jdbc.shaded.guava30. The logic is same with  [PR](https://github.com/apache/flink-connector-cassandra/pull/7) . WDYT? @MartijnVisser

[the remark this answers is not in the document]
Comment (zentol) [reply flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/utils/JdbcTypeUtil.java:29]: Like, `Lists.newArrayList` could just be replaced with `Arrays.asList`, whereas the ImmutableMap can be easily replaced by building a hashmap and wrapping it in Collections.unmodifiableMap.

[the remark this answers is not in the document]
Comment (WencongLiu) [reply flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/utils/JdbcTypeUtil.java:29]: Thanks for the reply😃 .I think your idea is right: not using Guava directly is one approach. Although Guava is currently not widely used in the repo, it cannot be ruled out that Guava will not be used in the future. In any case, I think both approaches are feasible. If you can reach a consensus on your views, I will implement it according to your views. @MartijnVisser @zentol

[the remark this answers is not in the document]
Comment (WencongLiu) [reply flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/utils/JdbcTypeUtil.java:29]: cc @MartijnVisser

[the remark this answers is not in the document]
Comment (WencongLiu) [reply flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/utils/JdbcTypeUtil.java:29]: @MartijnVisser, Could you please review this pull request again ? 😃

[the remark this answers is not in the document]
Comment (MartijnVisser) [reply flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/utils/JdbcTypeUtil.java:29]: > Although Guava is currently not widely used in the repo, it cannot be ruled out that Guava will not be used in the future.

I would prefer @zentol his approach. If we need Guava in the future, then let's introduce Guava in the future and remove it right now

[the remark this answers is not in the document]
Comment (WencongLiu) [reply flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/utils/JdbcTypeUtil.java]: Done. I've remove all guava dependencies in this repo. 😊 @MartijnVisser

**Review by MartijnVisser — APPROVED** [review]
LGTM

**Pull request merged**